src/app/src/components/FacilityListItemsFilterSearch.jsx

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useState } from 'react';
+import { func, string } from 'prop-types';
+
+import IconButton from '@material-ui/core/IconButton';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import TextField from '@material-ui/core/TextField';
+import Search from '@material-ui/icons/Search';
+import Close from '@material-ui/icons/Close';
+
+import { ENTER_KEYCODE, ESCAPE_KEYCODE } from '../util/constants';
+
+export default function FacilityListItemsTableSearch({
+    currentValue,
+    onSearch,
+}) {
+    const [searchText, setSearchText] = useState(currentValue);
+    const isCurrent = searchText === currentValue;
+
+    // Update internal state if the prop is updated
+    useEffect(() => setSearchText(currentValue), [currentValue]);
+
+    const handleSearch = () => {
+        if (isCurrent) {
+            return;
+        }
+
+        onSearch(searchText);
+    };
+
+    const handleTextChange = (e) => {
+        setSearchText(e.target.value);
+    };
+
+    const handleReset = () => {
+        setSearchText('');
+        onSearch('');
+    };
+
+    const handleTextKeyDown = (e) => {
+        if (e.keyCode === ENTER_KEYCODE) {
+            handleSearch();
+        } else if (e.keyCode === ESCAPE_KEYCODE) {
+            handleReset();
```

Comment:
In Chrome and Firefox on my macOS 10.14 workstation pressing the <kbd>esc</kbd> just makes the field loose focus.

![2019-05-31 16 09 21](https://user-images.githubusercontent.com/17363/58739385-ce51c100-83be-11e9-8fe8-3377b8f7b033.gif)

![2019-05-31 16 09 52](https://user-images.githubusercontent.com/17363/58739391-d27dde80-83be-11e9-9004-e1e13e8e1e58.gif)

In Safari it does clear the text box, but:
- If the search had not been applied the page reloads when it is not nessesary
- This does not match the behavior of pressing <kbd>esc</kbd> while the status dropdown has focus (which is removing focus from that field)

Consider dropping any special <kbd>esc</kbd> handling.

Reply:
Understood. Dropped in 37c518b.